.github/workflows/e2e.yml

```diff
@@ -0,0 +1,146 @@
+name: Build Data Image and Run End-to-end Tests
+
+on: push
+
+jobs:
+  e2e:
+    # Run in Sage repo on develop or main branches
+    # and on all branches in user-owned forks
+    if: ${{ github.ref_name == 'develop' || github.ref_name == 'main' || github.actor == github.repository_owner }}
```

Comment:
Forks do not have access to the Sage repo's secrets, so don't run this workflow in the Sage repo on pull_request trigger. Instead, run the workflow when there is a push to develop/main or when the user pushes to a repo that they own (i.e. their forked repo).